src/uct/ib/base/ib_iface.c

```diff
@@ -672,32 +679,88 @@ int uct_ib_iface_is_reachable(const uct_iface_h tl_iface,
                               const uct_device_addr_t *dev_addr,
                               const uct_iface_addr_t *iface_addr)
 {
-    uct_ib_iface_t *iface           = ucs_derived_of(tl_iface, uct_ib_iface_t);
-    int is_local_eth                = uct_ib_iface_is_roce(iface);
-    const uct_ib_address_t *ib_addr = (const void*)dev_addr;
-    uct_ib_address_pack_params_t params;
+    const uct_iface_is_reachable_params_t params = {
+        .field_mask         = UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR |
+                              UCT_IFACE_IS_REACHABLE_FIELD_IFACE_ADDR |
+                              UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING |
+                              UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING_LENGTH,
+        .device_addr        = dev_addr,
+        .iface_addr         = iface_addr,
+        .info_string        = NULL,
+        .info_string_length = 0
+    };
+    
+    return uct_ib_iface_is_reachable_v2(tl_iface, &params);
+}
 
-    uct_ib_address_unpack(ib_addr, &params);
+int uct_ib_iface_is_reachable_v2(const uct_iface_h tl_iface,
+                                 const uct_iface_is_reachable_params_t *params)
+{
+    const uct_device_addr_t *dev_addr = params->device_addr;
+    char* info_string                 = params->info_string;
```

Comment:
Add `const` where possible.

Reply:
The string buffer's context, which info_string points to, will be changed, so it wouldn't be used const to declare.